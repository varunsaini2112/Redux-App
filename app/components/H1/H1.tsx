import { PropsWithChildren } from "react";
import { Text } from "react-native";

import { fontStyles } from "../../config/appStyles";

const H1 = ({ children }: PropsWithChildren<{}>) => {
	return <Text style={fontStyles.h1}>{children}</Text>;
};

export default H1;
