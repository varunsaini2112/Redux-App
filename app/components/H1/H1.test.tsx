import { render, screen } from "@testing-library/react-native";
import H1 from "./H1";

describe("H1 Component", () => {
	it("should match snapshots", () => {
		const snapshot = render(<H1>Test Heading</H1>);

		expect(snapshot).toMatchSnapshot();
	});

	it("should render heading text", () => {
		render(<H1>Test Heading</H1>);

		expect(screen.getByText("Test Heading")).toBeOnTheScreen();
	});
});
