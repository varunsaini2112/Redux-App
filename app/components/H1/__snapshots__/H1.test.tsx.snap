// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Component should match snapshots 1`] = `
<Text
  style={
    {
      "color": "#132631",
      "fontSize": 24,
      "fontWeight": "600",
    }
  }
>
  Test Heading
</Text>
`;
